mlir/include/mlir/Interfaces/ValueBoundsOpInterface.h

```diff
@@ -316,6 +317,9 @@ class ValueBoundsConstraintSet {
 
   /// Builder for constructing affine expressions.
   Builder builder;
+
+  /// The current stop condition function.
+  StopConditionFn stopCondition = nullptr;
```

Comment:
Just wondering if this should be a `std::function` instead? `function_ref` begin non-owning could lead to some surprises. E.g. by doing `ValueBoundsConstraintSet cstr(..., /*stopCondition=*/[&]{ ... })`.

Reply:
Actually I ran into this exact issue in the past (in a different part of the codebase) and it was difficult to debug. Changed it to `std::function`.